src/openms/source/ANALYSIS/ID/ConsensusIDAlgorithmRanks.cpp

```diff
@@ -0,0 +1,103 @@
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Hendrik Weisser $
+// $Authors: Sven Nahnsen, Hendrik Weisser $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/ANALYSIS/ID/ConsensusIDAlgorithmRanks.h>
+#include <OpenMS/CONCEPT/LogStream.h>
+
+#include <cmath>
+#include <numeric> // for "accumulate"
+
+using namespace std;
+
+namespace OpenMS
+{
+  ConsensusIDAlgorithmRanks::ConsensusIDAlgorithmRanks()
+  {
+    setName("ConsensusIDAlgorithmRanks"); // DefaultParamHandler
+  }
+
+
+  void ConsensusIDAlgorithmRanks::preprocess_(
+    vector<PeptideIdentification>& ids)
+  {
+    // The idea here is that each peptide hit (sequence) gets assigned a score
+    // from each ID run, based on its rank in the list of search results.
+    // The best hit of a run will receive score 0, the second best 1, etc. up to
+    // a score of N = considered_hits - 1 for the last hit (if there are that
+    // many). A hit that was not observed in an ID run receives a score of
+    // N + 1 = considered_hits from that run. In the end the scores for each
+    // hit (sequence) are averaged and normalized to the range from 0
+    // (exclusive, worst) to 1 (inclusive, best).
+
+    current_number_of_runs_ = ((number_of_runs_ > 0) ? 
+                               number_of_runs_ : ids.size());
+    current_considered_hits_ = considered_hits_;
+    bool set_considered_hits = (considered_hits_ == 0);
+
+    for (vector<PeptideIdentification>::iterator pep_it = ids.begin();
+         pep_it != ids.end(); ++pep_it)
+    {
+      pep_it->assignRanks();
```

Comment:
ah ok, I see. `assignRanks` calls `sort` which takes the ordering into account.

However, it will still not really work if there are different score types, e.g. Mascot and SEQUEST scores that operate on different scales